packages/twenty-server/src/engine/core-modules/health/indicators/app.health.ts

```diff
@@ -20,10 +20,15 @@ export class AppHealthIndicator {
 
   async isHealthy(): Promise<HealthIndicatorResult> {
     const indicator = this.healthIndicatorService.check('app');
+    const SAMPLE_SIZE = 100;
 
     try {
       const workspaces = await this.objectMetadataService.findMany();
-      const workspaceIds = [...new Set(workspaces.map((w) => w.workspaceId))];
+      const allWorkspaceIds = [
+        ...new Set(workspaces.map((w) => w.workspaceId)),
+      ];
+
+      const workspaceIds = allWorkspaceIds.slice(0, SAMPLE_SIZE);
 
       const workspaceStats = await Promise.all(
```

Comment:
This is not efficient, you are doing 1 db request everytime you're in the loop. Always eager-load when you can.  Here the starting point should be pending migrations